dataprofiler/data_readers/json_data.py

```diff
@@ -276,36 +277,35 @@ def _load_data_from_file(self, input_file_path: str) -> List:
                 )
             return data
 
-    def _get_data_as_records(self, data: List) -> List[str]:
+    def _get_data_as_records(self, data: Union[pd.DataFrame, Dict, List]) -> List[str]:
         """
         Extract the data as a record format.
 
         :param data: raw data
         :type data: list
         :return: dataframe in record format
         """
-        _data: Union[pd.DataFrame, List]
-        _data = self._get_data_as_df(data)
-        _data = _data.to_dict(orient="records", into=OrderedDict)
-        for i, sample in enumerate(_data):
-            _data[i] = json.dumps(
+        data = self._get_data_as_df(data)
+        data = data.to_dict(orient="records", into=OrderedDict)
+        for i, sample in enumerate(data):
+            sample = self._convert_flat_to_nested_cols(sample)
+            data[i] = json.dumps(
                 self._convert_flat_to_nested_cols(sample), ensure_ascii=False
             )
-        return super(JSONData, self)._get_data_as_records(_data)
+        return super(JSONData, self)._get_data_as_records(data)
 
-    def _get_data_as_json(self, data: List) -> List[str]:
+    def _get_data_as_json(self, data: Union[pd.DataFrame, Dict, List]) -> List[str]:
         """
         Extract the data as a json format.
 
         :param data: raw data
         :type data: list
         :return: dataframe in json format
         """
-        _data: Union[pd.DataFrame, List]
```

Comment:
no longer having two variables of the data with _data
fixes type acceptance at the top